### ISSUE TITLE
[Site Title Block] Encourage users to make proper semantic decisions 

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#29644]
## What problem does this address?

Currently, with FSE, you can alter aspects of your site you couldn't previously including the Site Title. As a result as part of FSE Outreach Program testing, [the following feedback was raised](https://make.wordpress.org/test/2021/02/18/fse-program-testing-call-2-build-a-homepage-with-site-editing-blocks/#comment-1391):

>Also, I wasn’t sure about being able to choose h2-h6 and p for the site title. I think it is fairly common practice to set h1 on the front page, and div/span on other pages. If a person who is using FSE is a developer, then perhaps we can leave it to the developer to make the decisions (and should be aware of semantics and web standard). But if one is non-developer, then may be it is more friendly to limit the choice, or just have h1 and may be allow to choose which tag for non-front-page.

## What is your proposed solution?

Perhaps similar to how we show warnings for picking low contrast colors, we could show a nudge to encourage people to make good decisions around semantics for blocks like Site Title.

**Comment by carolinan** [comment]
Maybe the option could be moved the advanced section in the block settings sidebar, or vary depending on if it is inside a header template area or not.

**Comment by hedgefield** [comment]
I think in any situation it is ill-advised to make the title anything but an H1, right? Paging @afercia to confirm. If that's the case, let's remove that option completely, and leave the possibility of styling it to the text controls or advanced CSS?

**Comment by carolinan** [comment]
It is the other way around, for example, on a single post, the most important heading is normally the post title.
Sometimes the site title is at the bottom of the site footer, where it is not a heading over any type of content, then it should not be a semantic heading.

If it is a landing page without any other type of main heading, and the site title is in the site header, then it should be an H1.

**Comment by cbirdsong** [comment]
A common pattern I've seen and implemented is wrapping the site title/logo in an `<h1>` tag, but only on the home page where it makes sense as the page title. In traditional Wordpress terms it would look something like this:

```php
<? if (is_front_page()): ?><h1><? endif; ?>

<strong>
	<a href="<? echo esc_url( home_url() ); ?>" rel="home">
		
		<? echo wp_get_attachment_image(get_theme_mod( 'custom_logo' ), 'medium'); ?>
		<span <? if (has_custom_logo()) { ?>class="screen-reader-text"<? } ?>>
			<? bloginfo( 'name' ); ?>
		</span>
	</a>
</strong>

<? if (is_front_page()): ?></h1><? endif; ?>
```

Is there any affordance for this pattern in full-site editing?

**Comment by carolinan** [comment]
The block is not context aware, it does not know if it is on the home page or where it is on the page.

I am not sure if there are ny good solutions to this, because I also don't think the element should change automatically without the site creator being aware.

**Comment by carolinan** [comment]
As for making proper semantic decisions, the document outline will show if headings are in the incorrect order, but maybe it needs to be more visible.

**Comment by pagelab** [comment]
If there is no way for the user to decide on the title level according to context (homepage vs inner pages), then the site title probably shouldn't be a `<h1>` by default.

A [survey from 2017](https://webaim.org/projects/screenreadersurvey7/#heading) showed that the majority (66%) of screen reader users prefer only one main heading containing the document title. There's also a [Google News](https://support.google.com/news/publisher-center/answer/9607104?hl=en-GB) article that suggests the same as a best practice.

**Comment by carolinan** [comment]
Yes there is a way, and that is by adding different headers to different pages. That method It is not free from its own problems.

**Comment by pagelab** [comment]
> Yes there is a way, and that is by adding different headers to different pages. That method It is not free from its own problems.

I mean, an _automatic_ way to do this, as it is already possible with the Site Logo block by using `add_theme_support( 'custom-logo', [ 'unlink-homepage-logo' => true ] )`

**Comment by youknowriad** [comment]
It's not really clear how actionable is this issue. I'm going to close it for now but we can resurface it if there are clear rules we can apply.
